circuits/artifacts.go

```diff
@@ -0,0 +1,222 @@
+package circuits
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/vocdoni/vocdoni-z-sandbox/log"
+	"github.com/vocdoni/vocdoni-z-sandbox/types"
+)
+
+const downloadCircuitsTimeout = time.Minute * 5
+
+// BaseDir is where the artifact cache is expected to be found.
+// If the artifacts are not found there, they will be downloaded and stored.
+//
+// Defaults to ~/.cache/vocdoni-sequencer/zkCircuits/
+//
+// In any case, the hash of the asset will be appended at the end
+var BaseDir = func() string {
```

Comment:
I think this basedir should not be defined here. The data dir should be defined as a parameter of the NewCircuitsArtifacts constructor. Docker setups, tests, etc... might require a different basedir. 

In addition we will require datadir for the storage package and it makes sense to be the same.

Reply:
It can be overwritten like the `vocdoni-node` does [here](https://github.com/vocdoni/vocdoni-node/blob/2b98a4c1137163d0dafa656a4003d867c1612fb9/cmd/node/main.go#L370)

This works there for any environment (docker, test, etc...) and can be defined with the same root folder that the storage package

Reply:
I see. Well, I still think making this explicit in the constructor would make more sense.

Furthermore, I don't think a package library (different from main) should have specific OS details (i.e this does not work on Windows, and I'm not sure among Linux distributions and MacOS how standard is the home .cache directory).

But it's now up to you.

Reply:
I am agree, i defined the `BaseDir` for artifacts by default to `.cache/circuits-artifacts`. This variable can be overwritten by other packages. The creation of that folder (if it does not exists) is done during keys load and not in the `init()`.